server/src/internalClusterTest/java/org/elasticsearch/search/functionscore/QueryRescorerIT.java

```diff
@@ -983,4 +987,65 @@ private QueryBuilder fieldValueScoreQuery(String scoreField) {
             CombineFunction.REPLACE
         );
     }
+
+    public void testRescoreOnInnerHits() throws Exception {
```

Comment:
I think we would want to add a few other test cases, either here or in yaml tests

Reply:
💯 , this is the only test so far. I'll add lots more test cases once we confirm we want to move forward with this change.